### ISSUE TITLE
RFC: API Change: update LVGL from v7 to v8

### ISSUE THREAD

**Issue by brgl** [zephyrproject-rtos/zephyr#40901]
1. **Problem Description:**

Zephyr currently only supports LVGL v7. We'd like to upgrade it to v8 and drop v7 support as discussed during the dev-review meeting. Version 8 of LVGL brings in a lot of improvements over v7. v7 has not seen any releases since March 2021 while v8 is actively maintained.

2. **Proposed Change (detailed):**

We propose to upgrade the lvgl module to the latest v8 release and modify the glue code in zephyr (under ``zephyr/lib/gui/lvgl/``) to using v8. We'll also upgrade all lvgl examples that are in mainline zephyr but we don't guarantee that any user apps using lvgl will continue to work.

3. **Detailed RFC:**

As for the changes between v7 and v8 in the LVGL library itself - there are too many to list them here. Please refer to lvgl [changelog](https://github.com/lvgl/lvgl/blob/master/docs/CHANGELOG.md).
 
Currently Kconfig for lvgl is very fine-grained to the point where it allows to include separate widgets in the build. As widgets have undergone significant changes, the work will include reworking Kconfig options. These changes will break existing configs.

Initially we propose to keep the current structure of the LVGL glue code inside the zephyr repo. Once the initial conversion is done, it may be preferable to move it out of the main zephyr repo gradually (this part needs further research).
          
4. **Dependencies:**

This change has significant impact on out-of-tree users. In practice every lvgl application consisting of more than the simplest layouts will need to be reworked to use the v8 API.

Out-of-tree users should follow the lvgl [documentation](https://docs.lvgl.io/master/index.html) when porting applications. It's not possible to try and define any specific instructions here as the change affects an entire library.

**Comment by henrikbrixandersen** [comment]
API meeting:
- Please go ahead with the PR.
- Process for how to deal with external API changes will be dealt with in another issue.
